intel/customlists/module.go

```diff
@@ -0,0 +1,159 @@
+package customlists
+
+import (
+	"context"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/safing/portbase/modules"
+	"golang.org/x/net/publicsuffix"
+)
+
+var module *modules.Module
+
+const (
+	configModuleName  = "config"
+	configChangeEvent = "config change"
+)
+
+// Helper variables for parsing the input file.
+var (
+	isCountryCode      = regexp.MustCompile("^[A-Z]{2}$").MatchString
+	isAutonomousSystem = regexp.MustCompile(`^AS[0-9]+$`).MatchString
+)
+
+var (
+	filterListFilePath         string
+	filterListFileModifiedTime time.Time
+
+	parseLock sync.RWMutex
+)
+
+func init() {
+	module = modules.Register("customlists", prep, start, nil, "base")
+}
+
+func prep() error {
+	// register the config in the ui
+	err := registerConfig()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func start() error {
+	// register to hook to update after config change.
+	if err := module.RegisterEventHook(
+		configModuleName,
+		configChangeEvent,
+		"update custom filter list",
+		func(ctx context.Context, obj interface{}) error {
+			_ = checkAndUpdateFilterList()
+			return nil
+		},
+	); err != nil {
+		return err
+	}
+
+	// register timer to run every periodically and check for file updates
+	module.NewTask("intel/customlists file update check", func(context.Context, *modules.Task) error {
+		_ = checkAndUpdateFilterList()
+		return nil
+	}).Repeat(10 * time.Minute)
+
+	// parse the file at startup
+	_ = parseFile(getFilePath())
+	return nil
+}
```

Comment:
Looking at the task again, I noticed the idea was to always give the user a notification with stats about the loaded entries. So, we probably need to keep things this way.

Also, regarding the reloading, it would be nice to add an API endpoint to reload the list.